things/tracing/things.go

```diff
@@ -148,12 +148,19 @@ func (tcm thingCacheMiddleware) Remove(ctx context.Context, thingID string) erro
 }
 
 func createSpan(ctx context.Context, tracer opentracing.Tracer, opName string) opentracing.Span {
+	var span opentracing.Span
 	if parentSpan := opentracing.SpanFromContext(ctx); parentSpan != nil {
-		return tracer.StartSpan(
+		span = tracer.StartSpan(
 			opName,
 			opentracing.ChildOf(parentSpan.Context()),
 		)
+	} else {
+		span = tracer.StartSpan(opName)
 	}
 
-	return tracer.StartSpan(opName)
+	span.SetTag("span.kind", "client")
+	span.SetTag("peer.service", "postgres")
+	span.SetTag("db.type", "sql")
```

Comment:
I think that these tags shouldn't be present here, think about moving these tags to the `thingRepositoryMiddleware` and and `channelRepositoryMiddleware`.

Reply:
I actually moved these into the new dbMiddleware so that I have access to the query for a tag.  This way all of the tags are set in the same location.